racket/src/ChezScheme/s/cptypes.ss

```diff
@@ -1430,10 +1438,14 @@ Notes:
                       (predicate-implies? ret3 'bottom)) ;check bottom first
                  (values ir 'bottom pred-env-bottom #f #f)]
                 [(predicate-implies? ret2 'bottom) ;check bottom first
-                 (values (make-seq ctxt `(if ,e1 ,e2 ,void-rec) e3)
+                 (values (if (unsafe-unreachable? e2)
```

Comment:
I worry a little about whether we need to look deeper to find the `unreachable`, or propagate unreachability outward.

Reply:
IIUC, it seems the only easy case is `(begin (unsafe-unreachable) ...)`

Reply:
The tricky part for match is calling a function whose body is `(unsafe-unreachable)`.

Reply:
Would cptypes have already changed `(begin ($unreachable) ....)` in `e2` into `($unreachable)`, since `$unreachable` is marked as aborting?

Reply:
Right, it is already simplified to `($unreachable)`.

Reply:
A function whose body is `(unsafe-unreachable)` should be small enough to inline, so I think it is not a problem.

Reply:
Should we improve cp0 so that an application form`(e1 .... <aborting-call>  e2 ...)` turns into <aborting-call> (given that the parts of an application are allowed to be evaluated in any order)? And then that would take care of more `($unreachable)` cases?

Reply:
I prefer to keep the shallow version in the current proposal, and (later) make `cptypes` more aggressive to cut expressions when it detects an error.